Templates/Per Tenant Extension/.github/workflows/PullRequestHandler.yaml

```diff
@@ -25,7 +43,7 @@ env:
 
 jobs:
   PregateCheck:
```

Comment:
The `PregateCheck` exists to fail external PRs (from forks) that edit certain files.

With the new check the check will not be triggered for draft PRs, which is not desirable behavior.

Reply:
I don't see the sense in running status checks for PRs in draft mode. Even if the PR was editing blacklisted files, there's no risk of introducing breaking changes since it's not ready to be merged to begin with, right?

Reply:
Running status checks on draft have some advantages.

E.g. some developers prefer to have green build _before_ setting the PR as ready to review. Others, want to get feedback as soon as they start developing.

Since GitHub decided to run PR workflows even on draft PRs, we decided that there isn't anything to lose there.

Reply:
My motivation for this change is the fact that runtimes can be very long and I want to save as much compute time on GitHub's runners as possible. I understand that this can be mitigated by pushing as little as possible, but I'd like our developers to have the ability to keep their remote branches up to date without restrictions. Additionally, we have custom workflows that automate some processes by preparing pull requests, and those can end up in draft mode if manual input is required - any status checks would be pointless at this stage, so I'd like to avoid them.

Reply:
That's a valid point, but, as I said, that might not be aligned with what other developers what to experience.

So, generally you want to following scenarios to be supported:
1. When a draft PR is opened, no PR Build is triggered.
2. When a draft PR is converted to "ready for review", a PR Build is triggered.
3. When a non-draft PR is opened, a PR Build is triggered.

With your changes, 1. will not be satisfied, because GitHub doesn't differentiate between draft and non-draft PRs for the "opened" trigger. (Feel free to test that).
If a PR Build isn't triggered when the PR is opened, then how would you cover 3.?

Reply:
Oh, I didn't realize the `draft` parameter of a newly opened pull request always evaluates to false. In that case, it's impossible to prevent premature status checks.

Reply:
Maybe not prevent it, but make it not run the whole build. But that requires different effort.

Reply:
Would adding a draft condition to the build job suffice? Or are there other sections I need to adapt?

Reply:
Sorry, but I've tested again and GitHub does in fact differentiate between draft and non-draft PRs on the "opened" trigger.
The following workflow will be skipped if a new pull request is opened in draft mode:

```yml
name: Run Pre-Commit Hooks

on:
    pull_request:
        types: [opened, synchronize, reopened]
        branches:
            - main

jobs:
    pre-commit:
        name: Run Pre-Commit
        if: github.event.pull_request.draft == false
        runs-on: ubuntu-latest

        steps:
            - name: Checkout code
              uses: actions/checkout@v4
```

Reply:
Then how is it different from now?

> By default, a workflow only runs when a pull_request event's activity type is **opened**, **synchronize**, or **reopened**

Reply:
Can you elaborate a bit? Is the workflow skipped or is it never triggered?

My question was in regards of the triggers. If the default trigger types are _opened_, _synchronize_, or _reopened_, then why is the PR Build triggered on draft PRs in the first place?

Reply:
The triggers don't consider the status of the PR so the workflow is initiated all the same, but the jobs will be skipped because the `if`-condition isn't met.

Reply:
Oh, I see :)
That was the solution we would have gone with.